shared/modules/gas.utils.js

```diff
@@ -0,0 +1,109 @@
+ * @param {string} [options.maxFeePerGas] - The maximum fee in wei to pay per
+ *  gas used. maxFeePerGas is introduced in EIP 1559 and represents the max
+ *  total a user will pay per gas. Actual cost is determined by baseFeePerGas
+ *  on the block + maxPriorityFeePerGas. Value is hex string
+ * @returns {string} - The maximum total cost of transaction in hex wei string
+ */
+export function getMaximumGasTotalInHexWei({
+  gasLimit = '0x0',
+  gasPrice = '0x0',
+  maxFeePerGas,
+} = {}) {
+  if (maxFeePerGas && gasPrice) {
+    throw new Error(
+      `getMaximumGasTotalInHexWei expects either gasPrice OR maxFeePerGas but both were provided`,
+    );
+  }
+  if (maxFeePerGas) {
+    return addHexPrefix(
+      multiplyCurrencies(gasLimit, maxFeePerGas, {
+        toNumericBase: 'hex',
+        multiplicandBase: 16,
+        multiplierBase: 16,
+      }),
+    );
+  }
+  return addHexPrefix(
+    multiplyCurrencies(gasLimit, gasPrice, {
+      toNumericBase: 'hex',
+      multiplicandBase: 16,
+      multiplierBase: 16,
+    }),
+  );
+}
+
+/**
+ * Accepts an options bag containing gas fee parameters in hex format and
+ * returns a gasTotal parameter representing the minimum amount of wei the
+ * transaction will cost. For gasPrice types this is the same as max.
+ *
+ * @param {object} options - gas fee parameters object
+ * @param {string} [options.gasLimit] - the maximum amount of gas to allow this
+ *  transaction to consume. Value is a hex string
+ * @param {string} [options.gasPrice] - The fee in wei to pay per gas used.
+ *  gasPrice is only set on Legacy type transactions. Value is hex string
+ * @param {string} [options.maxFeePerGas] - The maximum fee in wei to pay per
+ *  gas used. maxFeePerGas is introduced in EIP 1559 and represents the max
+ *  total a user will pay per gas. Actual cost is determined by baseFeePerGas
+ *  on the block + maxPriorityFeePerGas. Value is hex string
+ * @param {string} [options.maxPriorityFeePerGas] - The maximum fee in wei to
+ *  pay a miner to include this transaction.
+ * @param {string} [options.baseFeePerGas] - The estimated block baseFeePerGas
+ *  that will be burned. Introduced in EIP 1559. Value in hex wei.
+ * @returns {string} - The minimum total cost of transaction in hex wei string
+ */
+export function getMinimumGasTotalInHexWei({
+  gasLimit = '0x0',
+  gasPrice,
+  maxPriorityFeePerGas,
+  maxFeePerGas,
+  baseFeePerGas,
+} = {}) {
+  if ((maxFeePerGas || maxPriorityFeePerGas || baseFeePerGas) && gasPrice) {
+    throw new Error(
+      `getMinimumGasTotalInHexWei expects either gasPrice OR the EIP-1559 gas fields, but both were provided`,
+    );
+  }
+  if (gasPrice) {
+    return getMaximumGasTotalInHexWei({ gasLimit, gasPrice });
+  }
+  const minimumFeePerGas = addCurrencies(baseFeePerGas, maxPriorityFeePerGas, {
+    toNumericBase: 'hex',
+    aBase: 16,
+    bBase: 16,
+  });
+  const minimum = addHexPrefix(
```

Comment:
Since we are possibly returning `getMaximumGasTotalInHexWei({ gasLimit, maxFeePerGas })` below, this might be unnecessarily computed. Maybe we could only initialize `minimum` in the case we are going to return it?

Reply:
ah yes, at one point i was comparing totals instead of the fee per gas, this is a carry over from that. Will fix it up now